src/wasm/kotlin/java/arcs/Entity.kt

```diff
@@ -144,41 +145,89 @@ class StringEncoder(private val sb: StringBuilder = StringBuilder()) {
     }
 }
 
-class TextDelegate {
-    private var isSet = false
-    private var v = ""
-    operator fun getValue(thisRef: Any?, property: KProperty<*>): String {
-        return v
-    }
+abstract class BaseEntity {
+    private var nextFieldId = 0
+    private val fieldSets = mutableSetOf<Int>()
+
+    internal fun registerField(delegate: EntityFieldDelegate<*>) {
+        val fieldId = nextFieldId++
+        delegate.onSet = {
+            fieldSets.add(fieldId)
 
-    operator fun setValue(thisRef: Any?, property: KProperty<*>, value: String) {
-        this.isSet = true
-        this.v = value
+            if (fieldSets.size == nextFieldId) {
+                isSet()
+            }
+        }
     }
+
+    open fun isSet() = Unit
 }
```

Comment:
If `Entity` is the base class, the contents of this could probably just live inside of it instead. Unless it's easier to test this way..

Reply:
Delegates went away :(